scripts/conflicts/diff.js

```diff
@@ -20,37 +26,90 @@ if (!server || argv.h || argv.help) {
+  writeStream.close();
+};
+
+DB.query('medic-conflicts/conflicts', {reduce:false})
   .then(conflicts => {
     console.log(`Found ${conflicts.rows.length} conflicts`);
 
+    const mainConflictFilesDirectory = './doc-conflicts';
     let ps = Promise.resolve();
+    if (!fs.existsSync(mainConflictFilesDirectory)){
+      try {
+        fs.mkdirSync(mainConflictFilesDirectory);
+      }catch(err){
+        console.log(`Error while creating main conflict folder: ${err}`);
+      }
+    }
     conflicts.rows.forEach(r => {
       ps = ps.then(() => {
-        // console.log(r);
-        if (r.key.length !== 1) {
-          console.log(`${r.id} has ${r.key.length} conflicts, we only want to deal with 1, skipping...`);
-        } else {
-          const docId = r.id;
-          const conflictRev = r.key[0];
-          return DB.get(docId, { open_revs: 'all' })
-            .then(results => {
-              // console.log(results);
-              // Make sure mine and theirs are the right way around
-              const mine = results.find(r => r.ok._rev !== conflictRev).ok;
-              const theirs = results.find(r => r.ok._rev === conflictRev).ok;
-
-              console.log('=====================');
-              console.log(r.id);
-              console.log('=====================');
-              console.log(jsondiff.diffString(mine, theirs));
-              console.log('\n\n');
-            });
-        }
+        const docId = r.id;
+        const arrayRevsIdThatShouldNotBeThere = r.key;
+        return DB.get(docId, { open_revs: 'all' })
+          .then(results => {
+
+            // Get the main_doc. Maindoc is the doc that isn't in the conflict list
+            const main_doc = results.filter(r => !arrayRevsIdThatShouldNotBeThere.includes(r.ok._rev))[0].ok; 
+
+            //Now we are going to put each doc, its conflicts and thier diff 
+            //in one file for manual inspection if required.
+            //The doc id might have characters that cannot be used to create folders.
+            let conflictDirectoryName = r.id;
+            if(r.id.includes('target')){
+              conflictDirectoryName = 'target_' + r.id.split(':')[0];
+            } else if (r.id.includes('task')){ //If doc is task add ask and part of doc id
+              conflictDirectoryName = 'task_' + r.id.split(':')[1];
+            } else {  // if it is just a contact doc use the doct Id to create folder
+              conflictDirectoryName = r.id;
+            }
+            
+            conflictDirectoryName.replace(/[^0-9a-z]/gi, '');
+            const conflictDirectoryPath = `${mainConflictFilesDirectory}/${conflictDirectoryName}`; 
+            if (!fs.existsSync(conflictDirectoryPath)){
+              try {
+                fs.mkdirSync(conflictDirectoryPath);
+              } catch(err){
+                console.log(err);
+              }
+            }
+
+            writeToFile(conflictDirectoryPath + '/' + 'mainDocument.json', main_doc);
```

Comment:
Please use `path.join` to generate paths instead of string concatenation: 
```suggestion
            writeToFile(path.join(conflictDirectoryPath, 'mainDocument.json'), main_doc);
```